extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -158,11 +158,11 @@ private function getTitleFromRequest(){
 		$articleId = $this->request->getInt( self::PARAM_ARTICLE_ID, NULL );
 		$articleTitle = $this->request->getVal( self::PARAM_ARTICLE_TITLE, NULL );
 
-		if ( !empty( $articleId ) && !empty( $articleTitle ) ) {
+		if ( !empty( $articleId ) && !is_null( $articleTitle ) && trim( $articleTitle ) !== '' ) {
```

Comment:
If my thinking is correct you could extract the new part to a public method:
```php
public function isTitleStringValid( $title ) {
  return !is_null( $title ) && trim( $title ) !== '';
}
```
then re-use this in both conditions:
```php
if ( !empty( $articleId ) && isTitleStringValid( $articleTitle ) ) {
  ...
}

if ( empty( $articleId ) && !isTitleStringValid( $articleTitle ) ) {
  ...
}
```
then you can even unit test `isTitleStringValid()`! :japanese_ogre: